src/shapes/Text/Text.ts

```diff
@@ -1766,18 +1766,9 @@ export class Text<
    * @static
    * @memberOf Text
    * @param {SVGElement} element Element to parse
-   * @param {Function} callback callback function invoked after parsing
    * @param {Object} [options] Options object
    */
-  static fromElement(
-    element: SVGElement,
-    callback: (text: Text | null) => any,
-    options: object
-  ) {
-    if (!element) {
-      return callback(null);
-    }
-
+  static async fromElement(element: SVGElement, options: object) {
     const parsedAttributes = parseAttributes(element, Text.ATTRIBUTE_NAMES),
       parsedAnchor = parsedAttributes.textAnchor || 'left';
     options = Object.assign({}, options, parsedAttributes);
```

Comment:
spread

Reply:
i shouldn't have digged in this function and dismiss as out of scope, i refactored most of it hopefully i didn't break it.
There is more to cut away there, but i just really wanted to expose those as promises.